pkg/rpcservice/middleware.go

```diff
@@ -43,7 +44,10 @@ func authorize(ctx context.Context) error {
 	}
 
 	if _, ok := md[requiredHeader]; !ok {
-		return status.Errorf(codes.Unauthenticated, fmt.Sprintf("%s header is not supplied", requiredHeader))
+		// Check for the alt header too, which is required for grpc web clients
+		if _, ok := md[altRequiredHeader]; !ok {
+			return status.Errorf(codes.Unauthenticated, fmt.Sprintf("%s header is not supplied", requiredHeader))
```

Comment:
nit: do you want to return the alt header in this case?
```suggestion
			return status.Errorf(codes.Unauthenticated, fmt.Sprintf("%s header is not supplied", altRequiredHeader))
```

Reply:
yep, good catch!